sdk/iot/core/inc/az_iot_hub_client.h

```diff
@@ -69,6 +69,10 @@ AZ_NODISCARD az_result az_iot_hub_client_init(
 
 /**
  * @brief Gets the MQTT user name.
+ * 
+ * The user name will be of the following format:
+ * [Format without module id] {iothubhostname}/{device_id}/?api-version=2018-06-30

```

Comment:
>2018-06-30 [](start = 72, length = 10)

I don't think we should hard-code this in comment.  We want to minimize of places we change this when we rev.

Also -- this is pretty old, is it really latest from hub that's universally deployed?

Reply:
This is straight from the documentation online [here](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-mqtt-support#using-the-mqtt-protocol-directly-as-a-device). I think it might be useful from the user perspective so that they know what to expect. If you think I should put it in one place only instead of in multiple places across the code base I'm open to that. (So would remove from the readme and `.c` file)

Reply:
That user documentation isn't great as it shouldn't do this either, because it locks customers onto a specific (very out of date) version.  This is really a variable just like {device_id}, except it comes from us.  I'm not sure how we define globals but I do think it should come from a .h, or historically it would at least.

---
In reply to: [393369267](https://github.com/Azure/azure-sdk-for-c/pull/455#discussion_r393369267) [](ancestors = 393369267)

Reply:
Ah I see your point. I should have linked this my bad.
https://github.com/Azure/azure-sdk-for-c/pull/455#discussion_r392545310
This has been noted and will be updated once we have a clear understanding about how versions/configs will be done. This API version will likely be a config in a header. For now Eric has said we will circle back and update. I do have a list of things to track for improvements. Let me know if that's alright with you.

Reply:
Yeah provided you're tracking an improvement, I don't consider my feedback to be a checkin blocker.